### PR TITLE
[WIP] Restart Rsyslog service when timezone changes

### PR DESCRIPTION
**Pull request by elemoine**
This is an attempt to fix a bug when the dates of logs in `/var/log/syslog` is not correct. This is WIP at the moment, and it requires more testing.

### CODE REVIEW

File: linux/system/init.sls [hunk 1]
```diff
@@ -60,3 +60,6 @@ include:
 {%- if system.haveged is defined %}
 - linux.system.haveged
 {%- endif %}
+{%- if system.rsyslog is defined %}
```
Comment (cznewt) [inline linux/system/init.sls]: nothing here

File: linux/system/rsyslog.sls [hunk 2]
```diff
@@ -0,0 +1,20 @@
+{%- from "linux/map.jinja" import system with context %}
+{%- if system.enabled %}
+
+{%- if system.rsyslog.enabled %}
```
Comment (cznewt) [inline linux/system/rsyslog.sls]: if system.syslog_service is defined

File: metadata/service/system/init.yml [hunk 3]
```diff
@@ -13,6 +13,8 @@ parameters:
           home: /root
       timezone: Europe/Prague
       cluster: default
+      rsyslog:
+        enabled: true
     network:
```
Comment (elemoine) [inline metadata/service/system/init.yml]: I took apparmor as an example here. https://github.com/tcpcloud/salt-formula-linux/blob/master/linux/system/apparmor.sls

**Comment by elemoine** [comment]
@cznewt does it look better to you now?

**Comment by elemoine** [comment]
I am closing this for now. Will reopen if necessary.

**Pull request closed without merging**